napari/components/add_layers_mixin.py

```diff
@@ -804,6 +814,20 @@ def _add_layers_with_plugins(
         # add each layer to the viewer
         added: List[layers.Layer] = []  # for layers that get added
         for data in layer_data:
+            # if user provided kwargs, use to override any meta dict values
+            # that were returned by the plugin
+            if kwargs:
+                layer_type = 'image' if len(data) < 3 else data[2]
+                valid_kwargs = prune_kwargs(kwargs, layer_type)
+                if len(data) == 1:
+                    data = (data[0], valid_kwargs)
+                elif len(data) > 1:
+                    assert isinstance(
+                        data[1], dict
+                    ), '2nd item should be a dict'
```

Comment:
Is this check necessary? Isn't it a guarantee of the type annotations that the 2nd item *will* be a dict?

Reply:
In fact, none of these lines inside `if kwargs` are covered by a test

Reply:
since those are returned by plugins, seemed like there was no guarantee of anything, so I figured a helpful error message there would be better.  can remove though if you prefer.  and will add a test (actually, probably will be useful to have a general `mock_plugin_return` test pattern)